instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -219,3 +231,19 @@ def _patched_api_call(self, original_func, instance, args, kwargs):
                 raise error
 
             return result
+
+    def _set_service_hooks(self, **kwargs):
+        self.request_hooks = dict()
+        self.response_hooks = dict()
+
+        request_hook_suffix = "_request_hook"
+        response_hook_suffix = "_response_hook"
+
+        for key, value in kwargs.items():
+            if key.endswith(request_hook_suffix):
+                service_name = key.split("_")[0].lower()
```

Comment:
Are you sure no service name includes an underscore?

Reply:
yes, this is the naming convention in boto:
```
client = boto3.client('apigatewaymanagementapi')
```
but I changed the way I extract the service name, and now it is up to the user to provide the right name